flow360/component/flow360_params/solvers.py

```diff
@@ -204,7 +204,10 @@ def to_solver(self, params, **kwargs) -> NavierStokesSolver:
 
         if self.low_mach_preconditioner:
             if self.low_mach_preconditioner_threshold is None:
-                self.low_mach_preconditioner_threshold = params.freestream.Mach
+                if hasattr(params.freestream, "Mach"):
+                    self.low_mach_preconditioner_threshold = params.freestream.Mach
+                else:
+                    self.low_mach_preconditioner_threshold = params.freestream.getMach()
```

Comment:
I think enough would be to just do:
```suggestion
                self.low_mach_preconditioner_threshold = params.freestream.to_solver(params, **kwargs).Mach                    
```
and other changes are not required. The `to_solver()` returns object that has Mach calculated

Reply:
but the issue is that Mach does not exist in all types of Freestream. Only after to_solver() we have access to Mach.

Reply:
right we need the value. Ignore my comment then.